controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -280,18 +280,29 @@ func (r *Reconciler) updateSpaceBinding(logger logr.Logger, spaceBinding *toolch
 
 func (r *Reconciler) createNewSpaceBinding(logger logr.Logger, memberCluster cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest, mur *toolchainv1alpha1.MasterUserRecord, space *toolchainv1alpha1.Space) error {
 	spaceBinding := spacebinding.NewSpaceBinding(mur, space, spaceBindingRequest.Name)
-	// set SBR labels on spacebinding
-	spaceBinding.Labels[toolchainv1alpha1.SpaceBindingRequestLabelKey] = spaceBindingRequest.GetName()
-	spaceBinding.Labels[toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey] = spaceBindingRequest.GetNamespace()
-	logger.Info("creating spacebinding", "spaceBinding.Name", spaceBinding.Name)
-	if err := r.setStatusProvisioning(memberCluster, spaceBindingRequest); err != nil {
-		return err
-	}
-	if err := r.Client.Create(context.TODO(), spaceBinding); err != nil {
-		return errs.Wrap(err, "unable to create SpaceBinding")
+	// check if there is already a SpaceBinding created with same name (there could be already a SpaceBinding for the same space and MUR generated by system admins for example)
+	existingSpaceBinding := &toolchainv1alpha1.SpaceBinding{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      spaceBinding.GetName(),
+	}, existingSpaceBinding)
```

Comment:
I would rather do the check somewhere else.
Before we go to this function we call
https://github.com/codeready-toolchain/host-operator/blob/b15e44ce05474814ad8388f76d639ee486b53e0e/controllers/spacebindingrequest/spacebindingrequest_controller.go#L99-L102

which Lists all existing SpaceBindings using the SpaceBindingRequest labels. 
https://github.com/codeready-toolchain/host-operator/blob/b15e44ce05474814ad8388f76d639ee486b53e0e/controllers/spacebindingrequest/spacebindingrequest_controller.go#L149-L156

instead of this, we should list them using the Space and MUR labels and then validate:
1. if there is no found, then let's create it
2. if there is one found, then check if it contains the SpaceBindingRequest labels. if not then return an error
3. if there are more than one found (which shouldn't really happen), then fail too (no matter who manages them) that it cannot properly apply the requested permissions because of multiple SBs to be present for the same combination of MUR & Space.

With this, we won't have to rely on the naming format to be the same everywhere (here as well as in sandbox-cli).

WDYT?

Reply:
yes that would work too, good point! Let me take a look and update it. 

I went this path because was the one requiring less code , but indeed not relying on the naming convention is a valid point!

Reply:
Please take a look at https://github.com/codeready-toolchain/host-operator/pull/898/commits/b53159786a5acc3d3df68ea0cc5eae3bbd1b5f0d when you have some time.